### PR TITLE
fix(settings): Split group lists

### PR DESCRIPTION
**Pull request by Pytal**
## Summary

- Requires https://github.com/nextcloud-libraries/nextcloud-vue/pull/5302
- Requires https://github.com/nextcloud-libraries/nextcloud-vue/pull/5347
- Requires NcVue release
- Adjust app navigation to use multiple lists

Before | After
--- | ---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by JuliaKirschenheuter — APPROVED** [review]
Great! Works ;)

**Review by emoral435 — APPROVED** [review]
This looks good to me 😀 is this potentially linked to one of the issues within the board?

**Comment by ShGKme** [comment]
I don't know if it is `@nextcloud/vue` or local issue, but scrolling is only applyed to the `NcAppNavigation` itself

scrolling

It makes groups least unreachable on a11y viewport size

image

**Comment by Pytal** [comment]
Good point @ShGKme :)

Removed the scrolling on purpose because `height: fit-content` with `overflow: auto` still makes it scroll and to keep the same design but I think adding a separate container which scrolls like `ul.app-navigation__list` when using default slot makes sense (in this case NcAppNavigationList)

image

**Comment by JuliaKirschenheuter** [comment]
I've tested it and it looks ok and fixes initial problem:

Screenshot from 2024-03-06 18-11-08

File: apps/files/src/views/Navigation.vue [hunk 1]
```diff
@@ -20,63 +20,67 @@
...
-				<!-- Child views if any -->
-				<NcAppNavigationItem v-for="child in childViews[view.id]"
-					:key="child.id"
-					:data-cy-files-navigation-item="child.id"
-					:exact-path="true"
-					:icon="child.iconClass"
-					:name="child.name"
-					:to="generateToNavigation(child)">
+	<Fragment>
+		<NcAppNavigation data-cy-files-navigation
```
Comment (ShGKme) [inline apps/files/src/views/Navigation.vue:24]: The change with Fragment is not needed after the revert in `@nextcloud/vue`

**Comment by ShGKme** [comment]
The bug on a small screen is not super critical, we can merge/backport the PR first and then updating on `@nextcloud/vue` should fix the layout for a small screen.

**Review by ShGKme — APPROVED** [review]

**Comment by skjnldsv** [comment]
Broke cypress :grin:

**Pull request merged**